### PR TITLE
fix/JiraIssue_IOPS-568

### PR DESCRIPTION
**Pull request by ryma2fhir**
https://nhsd-jira.digital.nhs.uk/browse/IOPS-568
Changed snomed values to reflect correct dm+d concept classes within SNOMED CT. Removed the dm+d system under compose as it seems redundant now the classes are added.

[VMPP](https://dd4c.digital.nhs.uk/dd4c/publishedmetadatas/intid/197)
[AMPP](https://dd4c.digital.nhs.uk/dd4c/publishedmetadatas/intid/187)
[AMP](https://dd4c.digital.nhs.uk/dd4c/publishedmetadatas/intid/186)
[VMP](https://dd4c.digital.nhs.uk/dd4c/publishedmetadatas/intid/196)
[VTM](https://dd4c.digital.nhs.uk/dd4c/publishedmetadatas/intid/198)
[ingredient](https://dd4c.digital.nhs.uk/dd4c/publishedmetadatas/intid/764) 

Have not expanded these valuesets as they have too many codes within them.

---

Once reviewed the following will need to be done:
Change the following from dm+d to SNOMED CT

   All medication must be represented using the NHS dm+d terminology.

   The code.coding.system must be https://dmd.nhs.uk/.
   The code.coding.code must be the NHS dm+d concept code.
   The code.coding.display must be the NHS dm+d concept description.

https://simplifier.net/guide/hl7fhirukcorer4release1/home/profilesandextensions/profileukcore-medication?version=current#code

Code within https://simplifier.net/guide/UK-Core-Implementation-Guide-1.0.0-STU1/Home/ProfilesandExtensions/ProfileUKCore-Immunization?version=current#vaccineCode

@SheenaSharma

### CODE REVIEW

File: valuesets/ValueSet-UKCore-AllergyCode.xml [hunk 1]
```diff
@@ -32,26 +32,22 @@
...
-		<include>
-			<system value="https://dmd.nhs.uk"/>
-		</include>
 		<include>
 			<system value="http://snomed.info/sct"/>
 			<filter>
 				<property value="constraint"/>
 				<op value="="/>
-				<value value="descendantOf 105590001 OR descendantOf 373873005 OR descendantOrSelfOf 716186003 OR 196461000000101 OR 196471000000108"/>
+				<value value="memberOf 999000581000001102 OR memberOf 999000561000001109 OR memberOf 999000541000001108 OR memberOf 999000791000001109 OR descendantOf 105590001 OR descendantOf 373873005 OR descendantOrSelfOf 716186003 OR 196461000000101 OR 196471000000108"/>
```
Comment (APageNHS) [inline valuesets/ValueSet-UKCore-AllergyCode.xml:50]: Can we add the following two concepts as descendentOrSelfOf
418634005 | Allergic reaction caused by substance (disorder) 
293104008 | Adverse reaction to component of vaccine product (disorder)

Comment (ryma2fhir) [reply valuesets/ValueSet-UKCore-AllergyCode.xml]: Both added.

**Comment by APageNHS** [comment]
I have discussed with RM about the fact we have removed the include dm+d, and shifted the dm+d groups over to the equivalent SNOMED CT refsets, and as a result, whether the first line of the description should be updated?

from: A set of codes from the following dm+d (dictionary of medicines and devices) concept classes

to something like : A set of codes from SNOMED CT reference sets representing the following dm+d (dictionary of medicines and devices) concept classes

But i'm not sure it is really needed, as I'd expect anyone using the UK Core for dm+d/SNOMED CT related stuff should have at least the basic understanding of the fact a dm+d APID/VPID is just a SNOMED CT concept id

RM stated that "I was thinking something along the same lines and came to the conclusion that the people using it should see it's a snomed refset and if they checked could verify it is the appropriate dm+d concept classes"

As such, I am happy that we have considered this impact, but both come to the conclusion it is an unnecessary change.

**Review by APageNHS — APPROVED** [review]
I have checked the refsets used, and the additions from the comment have been made. We have discussed whether wording needs adjusting and both agreed not to at this time.

All looks good

**Comment by ryma2fhir** [comment]
Needs to point to http://dmd.nhs.uk, so here are the new links from the [Ontoserver](https://ontoserver.csiro.au/shrimp/?concept=138875005&valueset=http%3A%2F%2Fsnomed.info%2Fsct%3Ffhir_vs&fhir=https%3A%2F%2Fontology.nhs.uk%2Fauthoring%2Ffhir)

[VMPP](https://dmd.nhs.uk/ValueSet/VMPP)
[AMPP](https://dmd.nhs.uk/ValueSet/AMPp)
[AMP](https://dmd.nhs.uk/ValueSet/AMP)
[VMP](https://dmd.nhs.uk/ValueSet/VMP)
[VTM](https://dmd.nhs.uk/ValueSet/VTM)
[ingredient](https://dmd.nhs.uk/ValueSet/INGREDIENT)

**Pull request merged**